pkg/serviceprovider/quay/quay.go

```diff
@@ -329,11 +320,13 @@ func (q *Quay) MapToken(ctx context.Context, binding *api.SPIAccessTokenBinding,
 func (q *Quay) Validate(ctx context.Context, validated serviceprovider.Validated) (serviceprovider.ValidationResult, error) {
 	ret := serviceprovider.ValidationResult{}
 
-	userPermissionAreaRequested := false
 	for _, p := range validated.Permissions().Required {
-		if p.Area == api.PermissionAreaUser && !userPermissionAreaRequested {
-			ret.ScopeValidation = append(ret.ScopeValidation, userRelatedPermissionsNotSupportedError)
-			userPermissionAreaRequested = true
+		switch p.Area {
```

Comment:
The idea is that only `PermissionAreaRegistry` and `PermissionAreaRegistryMetadata` are allowed. Every other permission will result in a validation error. The difference is that if the user, for example, specifies in binding 3x user area instead of one (which happens rarely, IMHO), he will get 3 validation errors saying that the user area is not supported. But the code is simpler, and I don't think it impacts the user experience.